CONTRIBUTING.md

```diff
@@ -42,29 +42,110 @@ Contributions to the source tree should follow the workflow described below:
    ```
 3. Build the image and test your changes.
 
-   Building and pushing the image to your own repository:
+   Before starting make sure you have your account created on [docker.io](https://www.docker.com/)and that you have logged-in from your terminal [docker login](https://docs.docker.com/engine/reference/commandline/login/).
+   First we are going to create the custom image with `make` utility (default one is [perconalab/percona-server-mysql-operator:<name-of-current-branch>](https://hub.docker.com/r/perconalab/percona-server-mysql-operator/).
+   Our build script (`e2e-tests/build`) disables caching and uses the experimental squash feature of Docker. To disable this behaviour and build the image use the following command:
 
    ```
-   make IMAGE=egegunes/percona-server-mysql-operator:k8sps-22
+   DOCKER_PUSH=0 DOCKER_SQUASH=0 DOCKER_NOCACHE=0 make IMAGE=<your-docker-id>/<custom-repository-name>:<custom-tag>
    ```
 
-   `Makefile` can automatically detect the image tag using the current branch. You can just override the image with:
+   Let's use `percona-server-mysql-operator` as an `custom-repository-name` and follow the previous example:
 
    ```
-   make IMAGE_TAG_BASE=egegunes/percona-server-mysql-operator
+   DOCKER_PUSH=0 DOCKER_SQUASH=0 DOCKER_NOCACHE=0 make IMAGE=egegunes/percona-server-mysql-operator:k8sps-22
```

Comment:
I don't think we should put this env vars to every command example, showing they're possible to override is enough